File: zig/src/indicators/core/adaptivity.zig
/// Classifies whether an indicator adapts its parameters to market conditions.
pub const Adaptivity = enum(u8) {
    /// Fixed parameters.
    static_ = 1,
    /// Adapts parameters to market conditions.
    adaptive = 2,

    pub fn asStr(self: Adaptivity) []const u8 {
        return switch (self) {
            .static_ => "static",
            .adaptive => "adaptive",
        };
    }
};

File: rs/src/indicators/core/adaptivity.rs
/// Classifies whether an indicator adapts its parameters to market conditions.
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
#[repr(u8)]
pub enum Adaptivity {
    /// Fixed parameters.
    Static = 1,
    /// Adapts parameters to market conditions.
    Adaptive = 2,
}

impl Adaptivity {
    pub fn as_str(&self) -> &'static str {
        match self {
            Self::Static => "static",
            Self::Adaptive => "adaptive",
        }
    }
}

impl std::fmt::Display for Adaptivity {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.write_str(self.as_str())
    }
}
